### PR TITLE
Convert Some Components From Class -> Functional

### PR DESCRIPTION
**Pull request by sigmachirality**
Converted a few of the simpler components from Class to Functional components, nothing major here.

### CODE REVIEW

File: frontend/src/NavBar.js [hunk 1]
```diff
@@ -1,59 +1,42 @@
...
-class NavBar extends Component {
-  constructor(props) {
-    super(props)
 
-    this.state = {
-      courseCount: Object.keys(localStorage).filter(a => !a.startsWith('meta-'))
-        .length,
+export const NavBar = () => {
+  const getCourseCount = () =>
+    Object.keys(localStorage).filter(a => !a.startsWith('meta-')).length
```
Comment (cacabo) [inline frontend/src/NavBar.js]: would be nice to have all localStorage operations be in a helper file called `store` where we can have `getSavedCourses` or something along those lines. Here's a TS rendition I have from a different project:

```ts
/**
 * Keys used in local storage
 */

const ACCESS_TOKEN_KEY = 'accessToken'
const REFRESH_TOKEN_KEY = 'refreshToken'

/**
 * Types relating to sessions
 */

interface IToken {
  email: string
  exp: number
  id: number
  type: EAccountType
}

let cachedStore: Storage

/**
 * Helper functions
 */

const getStore = (): Storage | undefined => {
  if (cachedStore) return cachedStore
  if (!isOnBrowser()) return undefined
  if (typeof window === 'undefined') return undefined
  if (!window || !window.localStorage) return undefined
  cachedStore = window.localStorage
  return cachedStore
}

/**
 * Getter functions
 */

export const getAccessToken = (): string | undefined => {
  const store = getStore()
  return (store && store.getItem(ACCESS_TOKEN_KEY)) || undefined
}

export const getRefreshToken = (): string | undefined => {
  const store = getStore()
  return (store && store.getItem(REFRESH_TOKEN_KEY)) || undefined
}

const getAccessTokenDecoded = (): IToken | undefined => {
  const accessToken = getAccessToken()
  return (accessToken && (jwt.decode(accessToken) as IToken)) || undefined
}

const getRefreshTokenDecoded = (): IToken | undefined => {
  const refreshToken = getRefreshToken()
  return (refreshToken && (jwt.decode(refreshToken) as IToken)) || undefined
}
```

Comment (cacabo) [reply frontend/src/NavBar.js]: If this is a path you want to take, this could be a different PR, as well

File: frontend/src/NavBar.js [hunk 2]
```diff
@@ -1,59 +1,42 @@
...
+  return (
+    <div id="header">
+      <span className="float-left">
+        <Link to="/" title="Go to Penn Course Review Home">
+          <div id="logo" />
+        </Link>
+        <SearchBar />
+      </span>
+      <span className="float-right">
+        <Link to="/cart" id="cart-icon" title="Course Cart">
```
Comment (cacabo) [inline frontend/src/NavBar.js]: Also would be good to make routes stored in a shared constants file, that can also be a different PR

File: frontend/src/ScoreTable.js [hunk 3]
```diff
@@ -1,87 +1,60 @@
...
+  const getTrProps = (_, rowInfo) => {
+    const { index, original, row } = rowInfo
+    const noRow = multi ? index in selected : index === selected
+    return rowInfo && row
+      ? {
+          onClick: () => {
+            if (multi) {
+              if (noRow) {
+                delete selected[index]
+              } else {
```
Comment (cacabo) [inline frontend/src/ScoreTable.js]: would prefer no nested ifs. Can you return early or combine clauses?

File: frontend/src/ScoreTable.js [hunk 4]
```diff
@@ -1,87 +1,60 @@
...
+                selected[index] = rowInfo
               }
+              onSelect && onSelect(selected)
```
Comment (cacabo) [inline frontend/src/ScoreTable.js]: I prefer if statements if we are calling functions / things with side effects. The and operators make it seem like this is just a boolean comparison when in reality state is changing

File: frontend/src/NavBar.js (hunk 1 above)
Comment (sigmachirality) [reply frontend/src/NavBar.js]: we're using localStorage for a lot of different things, so this would actually be really nice! I'll add a story for this.

File: frontend/src/ScoreTable.js [hunk 5]
```diff
@@ -1,87 +1,61 @@
...
 import ReactTable from 'react-table'
 
-/**
- * A wrapper for the react table with common table functionality and styling.
- */
-class ScoreTable extends Component {
-  constructor(props) {
-    super(props)
+const ScoreTable = props => {
+  const { noun, multi, data = [], onSelect = () => {} } = props
```
Comment (peterbaile) [inline frontend/src/ScoreTable.js]: can use object destructuring when passing in props, so instead of saying `props =>`, u can say `({ noun, multi, data = [], onSelect=()=>{} }) => `

Comment (sigmachirality) [reply frontend/src/ScoreTable.js]: can't, I need the entire props object to spread into the ReactTable component below
